### PR TITLE
Annotate settings.js flags that are relevant at compile time.

### PR DESCRIPTION
**Pull request by kripken**
99% of flags are link-only as they affect JS (or system libs). But a few affect compile as well.

Also remove a flag that is not used anywhere, SKIP_STACK_IN_SMALL.

cc @gabrielcuvillier

### CODE REVIEW

File: src/settings.js [hunk 1]
```diff
@@ -216,17 +236,11 @@ var DECLARE_ASM_MODULE_EXPORTS = 1;
 // Ignore closure warnings and errors (like on duplicate definitions)
 var IGNORE_CLOSURE_COMPILER_ERRORS = 0;
 
-// When enabled, does not push/pop the stack at all in functions that have no
-// basic stack usage. But, they may allocate stack later, and in a loop, this
-// can be very bad. In particular, when debugging, printf()ing a lot can exhaust
-// the stack very fast, with this option.  In particular, be careful with the
-// autodebugger! (We do turn this off automatically in that case, though.)
-var SKIP_STACK_IN_SMALL = 1;
-
```
Comment (sbc100) [inline src/settings.js]: Might make sense to make this a separate PR and add it to the changelog?    We could consider adding it to LEGACY_SETTINGS below as well.

**Comment by gabrielcuvillier** [comment]
Super great! Thank you for this clarification. I'm sure it will help people to port complex projects at some point in time, most notably allowing to do serious fine-tuning of settings.js flags without having to recompile everything :) 

Maybe STRICT=1 might be included as well, as it changes a few compiler defines (#undef EMSCRIPTEN + #define DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR), though I'm not sure this is really useful.

File: src/settings.js (hunk 1 above)
Comment (kripken) [reply src/settings.js]: Done.

**Comment by kripken** [comment]
Thanks @gabrielcuvillier, yeah, I forgot about the defines in STRICT, it does affect compile and link.

**Comment by VirtualTim** [comment]
Another +1 for this idea.
